server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -198,6 +198,33 @@ public long createOrUpdateSecret(String name, String encryptedSecret, String cre
     return ImmutableList.copyOf(results);
   }
 
+  /**
+   * @param name of secret series to look up secrets by.
+   * @param newestIdx the first index to select in a list of versions sorted by creation time
+   * @param oldestIdx the last index to select in a list of versions sorted by creation time
+   * @return Versions of a secret matching input parameters or Optional.absent().
+   */
+  public Optional<SecretSeriesAndVersions> getSecretVersionsByName(String name, int newestIdx,
+      int oldestIdx) {
+    checkArgument(!name.isEmpty());
+    checkArgument(newestIdx >= 0);
+    checkArgument(oldestIdx >= 0);
+
+    SecretContentDAO secretContentDAO = secretContentDAOFactory.using(dslContext.configuration());
+    SecretSeriesDAO secretSeriesDAO = secretSeriesDAOFactory.using(dslContext.configuration());
+
+    Optional<SecretSeries> series = secretSeriesDAO.getSecretSeriesByName(name);
+    if (series.isPresent()) {
+      long secretId = series.get().id();
+      Optional<ImmutableList<SecretContent>> contents =
+          secretContentDAO.getSecretVersionsBySecretId(secretId, newestIdx, oldestIdx);
```

Comment:
I've been experimenting with this, but it looks like we'd have to either take a generic Record from the query and process it into something useful (which could be hazardous if it isn't formatted as we expect) or define a new class to hold the information on the secret retrieved by the query, which I'm reluctant to do since we already have a lot of representations of secrets.  I'm not very familiar with the cost of queries in jOOQ--is avoiding an extra query worth these changes?